code/modules/mining/lavaland/weather.dm

```diff
@@ -46,7 +50,9 @@
 	update_areas()
 	for(var/mob/M in player_list)
 		if(M.z == target_z)
-			M << "<span class='danger'><B>[duration_message]</B></span>"
+			M << "<span class='userdanger'><i>[duration_message]</i></span>"
+			if(duration_sound)
+				M << duration_sound
```

Comment:
I've tested all the weather sounds. All of them play normally. If they're in a ruin or area where ash storms can't reach, they won't hear anything or see any messages. If they leave the area, the sound will continue, but they're all short enough that it could be viewed as simply leaving the storm for the shelter of another area.